src/shared/models/asset/asset.service.ts

```diff
@@ -2,12 +2,12 @@ import { Injectable } from '@nestjs/common';
 import { Blockchain } from 'src/blockchain/shared/enums/blockchain.enum';
 import { AssetRepository } from 'src/shared/models/asset/asset.repository';
 import { In } from 'typeorm';
-import { Asset } from './asset.entity';
+import { Asset, AssetType } from './asset.entity';
 
 export interface AssetQuery {
   dexName: string;
   blockchain: string;
-  isToken?: boolean;
+  type?: AssetType;
```

Comment:
Then we either need to default to Coin in getAssetByQuery (easy, temp solution) or actually make type mandatory too (harder fix, cause currently it is not always clear what type should we look for in calling contexts, like buy-crypto-batch and need some heavy refactoring)... wdyt?

Reply:
defaulting to Coin actually not an option, cause it anyways will require refactoring many calling contexts to explicitly ask for Token, when its needed.